doc/source/ray-core/objects/limiting-memory-usage.rst

```diff
@@ -0,0 +1,274 @@
+
+.. code-block:: bash
+
+    grep memory_monitor.cc /tmp/ray/session_latest/logs/raylet.out
+
+Which should print
+
+.. code-block:: bash
+
+    (raylet) memory_monitor.cc:45: MemoryMonitor initialized with usage threshold at 34664513536 bytes (0.97 system memory), total system memory bytes: 35738255360
+
+Memory usage threshold
+----------------------
+
+The memory usage threshold is used by the memory monitor to determine when it should start killing processes to free up memory. The threshold is controlled by two environment variables ``RAY_memory_usage_threshold_fraction`` (default: 0.9) and ``RAY_min_memory_free_bytes`` (default: 1 GiB). When the node starts it computes the usage threshold as follows:
+
+.. code-block:: bash
+
+    usage_threshold = max(system_memory * RAY_memory_usage_threshold_fraction, system_memory - RAY_min_memory_free_bytes)
+
+``RAY_min_memory_free_bytes`` can be disabled by setting its value to -1. In that case it only uses ``RAY_memory_usage_threshold_fraction`` to determine the usage threshold. Here we set the memory threshold to be 0.9 of system memory:
+
+.. code-block:: bash
+
+    ray stop; RAY_memory_monitor_interval_ms=250 RAY_memory_usage_threshold_fraction=0.9 RAY_min_memory_free_bytes=-1 ray start --head
+
+Now fetch the raylet logs:
+
+.. code-block:: bash
+
+    grep memory_monitor.cc /tmp/ray/session_latest/logs/raylet.out
+
+For a node with ~33 GiB of RAM the raylet log prints:
+
+.. code-block:: bash
+
+    (raylet) memory_monitor.cc:45: MemoryMonitor initialized with usage threshold at 32164429824 bytes (0.90 system memory), total system memory bytes: 35738255360
+
+On the other hand, if we set ``RAY_min_memory_free_bytes`` to a low value it will limit the amount of free memory reserved by the memory monitor. This helps limit the free memory for node with large amounts of RAM.
+
+.. code-block:: bash
+
+    ray stop; RAY_memory_monitor_interval_ms=250 RAY_memory_usage_threshold_fraction=0.5 RAY_min_memory_free_bytes=1000000000 ray start --head
+
+Now fetch the raylet logs again:
+
+.. code-block:: bash
+
+    grep memory_monitor.cc /tmp/ray/session_latest/logs/raylet.out
+
+For a node with ~33 GiB of RAM it should have a threshold close to 32GiB by leaving only 1000000000 bytes of free memory.
+
+.. code-block:: bash
+
+    (raylet) memory_monitor.cc:45: MemoryMonitor initialized with usage threshold at 34738255360 bytes (0.97 system memory), total system memory bytes: 35738255360
+
+Retry policy
+------------
+
+When a task or actor is killed by the memory monitor it will retry using a separate retry counter called the task oom retry instead of the typical retry that is :ref:`max_retries <task-fault-tolerance>` for Tasks and :ref:`max_restarts <actor-fault-tolerance>` for Actors. The number of oom retries is shared between tasks and actors and the default number is 15. It is configured by passing the environment variable ``RAY_task_oom_retries`` when starting Ray as well as the application.
```

Comment:
```suggestion
When a task or actor is killed by the memory monitor, it will retry using a separate retry counter called the task oom retry instead of the typical retry that is :ref:`max_retries <task-fault-tolerance>` for Tasks and :ref:`max_restarts <actor-fault-tolerance>` for Actors. The number of oom retries is shared between tasks and actors and the default number is 15. It is configured by passing the environment variable ``RAY_task_oom_retries`` when starting Ray as well as the application.
```